core/dataset.cc

```diff
@@ -968,6 +973,11 @@ generateRowsWhere(const SqlBindingScope & scope,
 
                             for (auto& c : inExpression->tuple->clauses) {
                                 ExpressionValue v = c->constantValue();
+
+                                //casting other types to string will give a different result than non-optimized path.
+                                if (!v.isString())
```

Comment:
Effectively we are breaking this, because it was working *only* in some optimized case and not in the generic case - giving inconsistent result.